### DISCUSSION TITLE
Toobar items not vertically centered

### DISCUSSION THREAD

**Discussion by iDevelopper** [LeoNatan/LNPopupController#542] [General] [answered]
Hello,

How can we successfully center vertically the toolbar buttons (case of system symbols) ?

So that they are vertically aligned with the titles view too?

**Comment by LeoNatan** [comment]
You can take a look at how I add them in the demo app. System images in toolbars and navigation bars are quite finicky.

**Comment by iDevelopper** [comment]
That's what I did. In your example the buttons are not centered vertically. Look carefully.

**Reply by LeoNatan** [reply]
Can you post a screenshot?

**Reply by LeoNatan** [reply]
![IMG_5825](https://github.com/LeoNatan/LNPopupController/assets/2270433/6334d68d-f00b-4156-9e12-70f856bc2c34)

Looks centered to me.

**Comment by iDevelopper** [comment]
![Capture d’écran   2023-09-17 à 06 09 39](https://github.com/LeoNatan/LNPopupController/assets/1782450/f15bcaf3-c0c4-4845-90ca-9396f3a834e4)

**Comment by iDevelopper** [comment]
In fact it is the imageView and the titlesView which are centered in relation to the contentView and not in relation to the toolbar. And the contentView is bigger than the toolbar.
<img width="680" alt="Capture d’écran 2023-09-17 à 06 35 29" src="https://github.com/LeoNatan/LNPopupController/assets/1782450/49acfa49-8229-473f-ad5b-f69fd4d4c71f">

**Reply by LeoNatan** [reply]
That’s an implementation detail. The titles container is always full height of the popup bar, but the toolbar is always centered in the content view. 

https://github.com/LeoNatan/LNPopupController/blob/43e71be8e7b802f7d1d0bc9a8cc62f036cd7c8d7/LNPopupController/LNPopupController/Private/LNPopupBar.m#L437

**Reply by LeoNatan** [reply]
I don’t remember why there is one point offset there, but it probably looked better to me at some point. 🤣

**Reply by iDevelopper** [reply]
I don't know ! I don't understand why the contentView is larger than the toolbar actually.

**Reply by LeoNatan** [reply, marked as answer]
The toolbar height must be left to the system layout. If you notice, I don’t touch the height of the toolbar. I just center it.
